File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 4/21/2026, 11:25:43 PM
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        for(int i = n - 1; i >= 0;i--){
            if(nums[i] == 0){
                for(int j = i; j < n - 1 ; j++){
                    nums[j] = nums[j+1];
                }
                nums[n-1] = 0;
            }
        }
    }
}

File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 8/21/2026, 11:31:40 AM
1class Solution(object):
2    def moveZeroes(self, nums):
3        """
4        :type nums: List[int]
5        :rtype: None Do not return anything, modify nums in-place instead.
6        """
7        i = 0
8        k = 0
9        while i < len(nums):
10            if nums[i] != 0:
11                nums[k], nums[i] = nums[i], nums[k]
12                k+=1
13            i+=1
14        return nums
15